### PR TITLE
fix: add check for CodeUri property type

### PR DESCRIPTION
**Pull request by paulhcsun**
### Description of changes
Add check on CodeUri property type and throw error if something other that dict or string is passed in
### Description of how you validated changes
Transformation test
### Checklist

- [x] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [x] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [x] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/model/s3_utils/uri_parser.py [hunk 1]
```diff
@@ -85,17 +85,7 @@ def construct_s3_location_object(
 
         s3_pointer = location_uri
 
-    else:
```
Comment (ConnorRobertson) [inline samtranslator/model/s3_utils/uri_parser.py:88]: Why was this removed?

Comment (paulhcsun) [reply samtranslator/model/s3_utils/uri_parser.py:88]: oh I think this was when I resolved merge conflicts here. One sec

**Review by ssenchenko — APPROVED** [review]

**Review by moelasmar — APPROVED** [review]
